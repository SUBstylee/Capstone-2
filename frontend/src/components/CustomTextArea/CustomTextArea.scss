.CustomTextArea{
    position: relative;
    margin: 20px 0 40px;
    width: 300px;
    textarea{
        background-color: #fff !important;
        border: none;
        border-bottom: 2px #919191 solid;
        border-top: 2px #919191 solid;
        display: block;
        width: 100%;
        padding: 15px 0;
        font-size: 20px;
    };

    textarea:focus, textarea:valid{
        outline: 0;
        border-bottom-color: rgb(28, 28, 165);
        border-top-color: rgb(28, 28, 165);
    };

    textarea:-webkit-autofill,
    textarea:-webkit-autofill:hover, 
    textarea:-webkit-autofill:focus, 
    textarea:-webkit-autofill:active{
        -webkit-box-shadow: 0 0 0 30px white inset !important;
    }

    label{
        position: absolute;
        top:5px;
        left: 0;
    }
    label span{
        display: inline-block;
        font-size: 18px;
        min-width: 5px;
        transition: 0.3s cubic-bezier(0.68,-0.55,0.265,1.55);
    }
    textarea:focus + label span,
    textarea:valid + label span{
        color: rgb(28, 28, 165);
        transform: translateY(-35px);
    }
};