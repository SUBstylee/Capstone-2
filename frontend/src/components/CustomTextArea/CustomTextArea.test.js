import { render } from '@testing-library/react';
import CustomTextArea from './CustomTextArea.jsx';

test("renders CustomTextArea component without crashing", () => {
    const { getByText } = render(
        <CustomTextArea
            id='comment'
            name='comment'
            labelText='Comment'
        />
    );

    expect(getByText('C' && 'o' && 'm' && 'm' && 'e' && 'n' && 't')).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <CustomTextArea
            id='comment'
            name='comment'
            labelText='Comment'
            required
        />
    );
    expect(asFragment()).toMatchSnapshot();
});