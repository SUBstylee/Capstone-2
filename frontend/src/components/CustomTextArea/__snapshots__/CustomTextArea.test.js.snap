// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CustomTextArea"
  >
    <textarea
      id="comment"
      name="comment"
      required=""
    />
    <label
      for="comment"
    >
      <span
        style="transition-delay: 0ms;"
      >
        C
      </span>
      <span
        style="transition-delay: 50ms;"
      >
        o
      </span>
      <span
        style="transition-delay: 100ms;"
      >
        m
      </span>
      <span
        style="transition-delay: 150ms;"
      >
        m
      </span>
      <span
        style="transition-delay: 200ms;"
      >
        e
      </span>
      <span
        style="transition-delay: 250ms;"
      >
        n
      </span>
      <span
        style="transition-delay: 300ms;"
      >
        t
      </span>
    </label>
  </div>
</DocumentFragment>
`;
